js/src/workspaces/slot.js

```diff
@@ -171,11 +172,31 @@
 
           _this.eventEmitter.publish('ADD_WINDOW', windowConfig);
 
-        } else if (typeof imageInfoUrl !== 'undefined') {
+        } 
+        
+        else if (typeof imageInfoUrl !== 'undefined') {
           if (!_this.state.getStateProperty('manifests')[imageInfoUrl]) {
             _this.eventEmitter.publish('ADD_MANIFEST_FROM_URL', imageInfoUrl, "(Added from URL)");
           }
-        } else {
+        } 
+        else if (typeof collectionUrl !== 'undefined'){
+          jQuery.getJSON(collectionUrl).done(function (data, status, jqXHR) {
+            if (data.hasOwnProperty('manifests')){
+              jQuery.each(data.manifests, function (ci, mfst) {
+                if (!_this.state.getStateProperty('manifests')[imageInfoUrl]) {
+                  _this.eventEmitter.publish('ADD_MANIFEST_FROM_URL', mfst['@id'], "(Added from URL)");
+                }
+              });
+            }
+          });
+
+          //TODO: 
+          //this works;
+          //but you might want to check if some "publish" action would be better
+          _this.addItem();
```

Comment:
Good point, but I think this will be okay for now. We need to think about how the different top-level screens can be represented in the data structure, and how to update them accordingly without calling the methods of things. 

Any update needs to ultimately be done through an "action", so that the state manager can send out notifications to subscribers(UI components) that depend on that state.